ui/src/app/ui-banner/ui-banner.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+require('./ui-banner.scss');
+
+export interface UiBannerProps extends React.Props<any> {
+    items: {apiUrl:string, fields: Array<string> };
+}
+
+
+export class Banner extends React.Component<UiBannerProps>{
+    state:any = {
+        bannerVisible:false,
+        isLoaded:false,
+        apiData:[]
+    }
+    constructor(props:UiBannerProps){
+        super(props)
+    }
+    setBanner = () => {
+        this.setState({
+        bannerVisible:false,
```

Comment:
If visibility is stored in the react state then user will have to hide banner every time after page is reloaded. We can leverage user preference state that stores user settings in browser local storage:

https://github.com/argoproj/argo-cd/blob/c7dbe4883b040d1ee2850760698ad1831c65e59f/ui/src/app/shared/services/view-preferences-service.ts#L25-L33

I suggest adding a new field `isBannerHidden`.  We could use `boolean` data type but in this case it is unclear how to make sure user sees second announcement after first one is hidden. So instead of boolean true/false lets store the hash of banner message ( hash can be just md5 hash of all concatenated banner text fields ). So if hash stored in `isBannerHidden` matches to current hash of banner then it is considered hidden.

Reply:
Thanks for the nice feedback @alexmt will address these issues.